demo/info_section.js

```diff
@@ -198,7 +197,11 @@ shakaDemo.updateLanguageOptions_ =
   // Populate list with new options.
   languages.forEach(function(lang) {
     var option = document.createElement('option');
-    option.textContent = lang;
+    var currentTrack = tracks.filter(function(track) {
```

Comment:
There are two things wrong with this part:

1. currentTrack is a misleading name.  It is not the currently-selected track you've identified, it's just any track that matches the language.
2. You shouldn't annotate the languages with track labels.  Instead, annotate the track list.

For example, if you have two tracks of the same language, as @erankor suggested is common for this feature, you would have these tracks:

```js
[
  { id: 1, lang: 'en', label: 'with music' },
  { id: 2, lang: 'en', label: 'without music' },
]
```

But the only language in the language list will be "en".  With what you've written, your language options in HTML would wind up as:

```html
<select>
<option value="en">with music</option>
</select>
```

While the variant tracks would still both show "language: en" with no differentiation.

I think the method you really want to modify here is `shakaDemo.updateTrackOptions_`, specifically the `formatters` callbacks.